prow/plugins/label-based-commenting/label-based-commenting.go

```diff
@@ -0,0 +1,286 @@
+		return err
+	}
+	return handle(pc.Logger, pc.GitHubClient, cp, pc.PluginConfig.LabelBasedCommenting, e)
+}
+
+func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
+	if !handlePRActions[pre.Action] {
+		return nil
+	}
+	e := &rblm.Event{
+		Org:    pre.Repo.Owner.Login,
+		Repo:   pre.Repo.Name,
+		Number: pre.PullRequest.Number,
+		Branch: pre.PullRequest.Base.Ref,
+		Author: pre.PullRequest.User.Login,
+		Label:  pre.Label.Name, // This will be empty for non-label events.
+		IsPR:   true,
+	}
+	cp, err := pc.CommentPruner()
+	if err != nil {
+		return err
+	}
+	return handle(pc.Logger, pc.GitHubClient, cp, pc.PluginConfig.LabelBasedCommenting, e)
+}
+
+// matchingConfigs filters irrelevant RequireMtchingLabel configs from
+// the list of all configs.
+func matchingConfigs(org, repo, branch, label string, allConfigs []plugins.LabelBasedCommenting) []plugins.LabelBasedCommenting {
+	var filtered []plugins.LabelBasedCommenting
+	for _, cfg := range allConfigs {
+		if rblm.ShouldConsiderConfig(org, repo, branch, label, cfg.RegexBasedLabelMatch) {
+			filtered = append(filtered, cfg)
+		}
+	}
+	return filtered
+}
+
+func handleCommentEvent(pc plugins.Agent, ce github.GenericCommentEvent) error {
+	// Only consider open PRs and new comments.
+	if ce.IssueState != "open" || ce.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	cp, err := pc.CommentPruner()
+	if err != nil {
+		return err
+	}
+
+	return handleComment(pc.Logger, pc.GitHubClient, cp, pc.PluginConfig.LabelBasedCommenting, &ce)
+}
+
+func handleComment(log *logrus.Entry, ghc rblm.GithubClient, cp rblm.CommentPruner, configs []plugins.LabelBasedCommenting, e *github.GenericCommentEvent) error {
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	number := e.Number
+
+	event := &rblm.Event{
+		Org:    org,
+		Repo:   repo,
+		Number: number,
+		Author: e.User.Login,
+		IsPR:   e.IsPR,
+	}
+	if e.IsPR {
+		pr, err := ghc.GetPullRequest(org, repo, number)
+		if err != nil {
+			return err
+		}
+		event.Branch = pr.Base.Ref
+	}
+	return handle(log, ghc, cp, configs, event)
+}
+
+func handle(log *logrus.Entry, ghc rblm.GithubClient, cp rblm.CommentPruner, configs []plugins.LabelBasedCommenting, e *rblm.Event) error {
+	// Find any configs that may be relevant to this event.
+	matchConfigs := matchingConfigs(e.Org, e.Repo, e.Branch, e.Label, configs)
+	if len(matchConfigs) == 0 {
+		return nil
+	}
+
+	err := rblm.LabelPreChecks(e, ghc, extractRegexBasedLabelMatchConfigs(configs))
+	if err != nil {
+		return err
+	}
+
+	// Handle the potentially relevant configs.
+	for _, cfg := range matchConfigs {
+		hasMatchingLabel := false
+		for _, label := range e.CurrentLabels {
+			hasMatchingLabel = hasMatchingLabel || cfg.Re.MatchString(label.Name)
+		}
+
+		commentResponse := plugins.FormatSimpleResponse(e.Author, constructCommentResponse(e, cfg.Comment))
+		if hasMatchingLabel {
+			// TODO(MadhavJivrajani): we need to check if the comment has already
```

Comment:
It would be nice if this could be done before merging this PR. We really want to avoid comment-spam